rest/rest-cli/src/test/java/org/ow2/proactive_grid_cloud_portal/cli/cmd/sched/ListJobCommandTest.java

```diff
@@ -0,0 +1,228 @@
+    }
+
+    @Test
+    public void testAllJobsReturned() throws Exception {
+
+        int index = 0;
+        int offset = 10;
+
+        List<UserJobData> userJobInfoList = createUserJobInfoList(10);
+
+        RestPage<String> restPage = new RestPage(userJobInfoList, 1);
+        Map<Long, List<UserJobData>> stateMap = new HashMap<>();
+        stateMap.put(1L, userJobInfoList);
+        RestMapPage<Long, ArrayList<UserJobData>> page = new RestMapPage(stateMap, 1);
+
+        Mockito.when(schedulerRestInterfaceMock.jobs("sessionid", index, offset)).thenReturn(restPage);
+        Mockito.when(schedulerRestInterfaceMock.revisionAndJobsInfo("sessionid", index, 10, false, true, true, true))
+               .thenReturn(page);
+
+        new ListJobCommand("limit=" + offset, "from=" + index).execute(currentContextMock);
+
+        Mockito.verify(schedulerRestInterfaceMock).revisionAndJobsInfo("sessionid", index, 10, false, true, true, true);
+    }
+
+    @Test
+    public void testZeroJobsReturned() throws Exception {
+
+        int index = 0;
+        int offset = 1;
+
+        List<UserJobData> userJobInfoList = createUserJobInfoList(1);
+
+        RestPage<String> restPage = new RestPage(userJobInfoList, 1);
+        Map<Long, List<UserJobData>> stateMap = new HashMap<>();
+        stateMap.put(1L, userJobInfoList);
+        RestMapPage<Long, ArrayList<UserJobData>> page = new RestMapPage(stateMap, 1);
+
+        Mockito.when(schedulerRestInterfaceMock.jobs("sessionid", index, offset)).thenReturn(restPage);
+        Mockito.when(schedulerRestInterfaceMock.revisionAndJobsInfo("sessionid", index, 1, false, true, true, true))
+               .thenReturn(page);
+
+        new ListJobCommand("limit=" + offset, "from=" + index).execute(currentContextMock);
+
+        Mockito.verify(schedulerRestInterfaceMock).revisionAndJobsInfo("sessionid", index, 1, false, true, true, true);
+
+    }
+
+    @Test
+    public void testOneJobReturned() throws Exception {
+
+        int index = 1;
+        int offset = 1;
+
+        List<UserJobData> userJobInfoList = createUserJobInfoList(3);
+
+        RestPage<String> restPage = new RestPage(userJobInfoList, 1);
+        Map<Long, List<UserJobData>> stateMap = new HashMap<>();
+        stateMap.put(1L, userJobInfoList);
+        RestMapPage<Long, ArrayList<UserJobData>> page = new RestMapPage(stateMap, 1);
+
+        Mockito.when(schedulerRestInterfaceMock.jobs("sessionid", index, offset)).thenReturn(restPage);
+        Mockito.when(schedulerRestInterfaceMock.revisionAndJobsInfo("sessionid", index, 3, false, true, true, true))
+               .thenReturn(page);
+
+        new ListJobCommand("limit=" + offset, "from=" + index).execute(currentContextMock);
+
+        Mockito.verify(schedulerRestInterfaceMock).revisionAndJobsInfo("sessionid", index, 3, false, true, true, true);
+    }
+
+    @Test
+    public void testTwoJobsReturned() throws Exception {
+
```

Comment:
We provide whether the index and offset or the limit (max number of jobs), and we retrieve the jobs. Those arguments match the position of jobs in the list, and not their IDs.